### ISSUE TITLE
FNX3-16744 ⁃ [Bug] Cold start of Fenix crashes the app with a new intent

### ISSUE THREAD

**Issue by jonalmeida** [mozilla-mobile/fenix#3032]
## Steps to reproduce
1. Ensure Fenix is your default browser.
1. Send a tab your device.
1. Click the tab notification.
1. Wait for the tab to open.

Alt STR:

1. Ensure Fenix is your default browser.
1. Ensure the app is closed.
1. Click a link somewhere to be opened in Fenix.

### Expected behavior
* The tab loads.

### Actual behavior
* Crash:

```
2019-05-30 21:48:16.958 5820-5855/org.mozilla.fenix.debug E/WM-WorkerWrapper: Work [ id=0492cd7d-e6d8-415a-a193-6cbe7b78f449, tags={ Immediate, mozilla.components.feature.sync.WorkManagerSyncWorker, Common } ] failed because it threw an exception/error
    java.util.concurrent.ExecutionException: kotlin.KotlinNullPointerException
        at androidx.work.impl.utils.futures.AbstractFuture.getDoneValue(AbstractFuture.java:516)
        at androidx.work.impl.utils.futures.AbstractFuture.get(AbstractFuture.java:475)
        at androidx.work.impl.WorkerWrapper$2.run(WorkerWrapper.java:289)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:919)
     Caused by: kotlin.KotlinNullPointerException
        at mozilla.components.feature.sync.WorkManagerSyncWorker.doWork(WorkManagerSyncDispatcher.kt:223)
        at androidx.work.CoroutineWorker$startWork$1.invokeSuspend(CoroutineWorker.kt:66)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(Dispatched.kt:238)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:594)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.access$runSafely(CoroutineScheduler.kt:60)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:742)
```

### Device information
* Android device: Pixel 2 (Android 9)
* Fenix version: ab39a508

cc: @grigoryk

**Comment by jonalmeida** [comment]
Opening the app after a while (overnight since last open) caused this which might be a more useful stacktrace (see [Sentry FENIX-1FT](https://sentry.prod.mozaws.net/operations/fenix/issues/5860050/)):
```
kotlin.KotlinNullPointerException: null
    at mozilla.components.feature.sync.GeneralSyncManager.syncNow(BackgroundSyncManager.kt:7)
    at org.mozilla.fenix.HomeActivity$onResume$1.invokeSuspend(HomeActivity.kt:13)
    at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:2)
    at kotlinx.coroutines.DispatchedTask.run(Dispatched.kt:15)
    at android.os.Handler.handleCallback(Handler.java:873)
    at android.os.Handler.dispatchMessage(Handler.java:99)
    at android.os.Looper.loop(Looper.java:193)
    at android.app.ActivityThread.main(ActivityThread.java:6718)
    at java.lang.reflect.Method.invoke(Method.java)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
```

**Comment by jonalmeida** [comment]
It looks like the `syncDispatcher` is null which is what the trace is referring to.

**Comment by jonalmeida** [comment]
I have a general fix for this by adding the appropriate null checks that I'm verifying now and putting a PR up for in a-c.

**Comment by jonalmeida** [comment]
@grigoryk I see now from the tests that the null pointer exception is intentional if something happened that shouldn't have. I think I'm at a road block while I try to figure this out.

**Comment by jonalmeida** [comment]
With limited understanding still: maybe we can check the account first before calling sync now?

**Comment by jonalmeida** [comment]
This is fixed now that the a-c issue is closed.

**Comment by AndiAJ** [comment]
Hi, **verified as fixed** on the latest Fenix build (#11790617 from 28/6) using the following devices:

• Google Pixel 3XL (Android 9)
• Huawei Mate 20 Lite (Android 8.1.0)
• Samsung Galaxy S7 (Android 7.0)
